calcom/cal.com
# Fix: Handle Error for Google Calendar Authorization Cancellation

## What does this PR do?

> This PR adds error handling for when users cancel Google Calendar authorization, redirecting them to a safe page with an error message instead of a 500 Internal Server Error.

- Fixes #15954

> I have self-reviewed the code, ensuring it handles errors correctly and adheres to coding standards.

> I have added a Docs issue in the code (https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a documentation change

## How should this be tested?

> Here’s a detailed answer for testing the changes:

- Test by navigating to the internal website with the GcalConnect atom, clicking "Connect Google Calendar," selecting an account, and then canceling the authorization. Verify that the user is redirected to a safe page and that no 500 Internal Server Error occurs.

- - GOOGLE_CLIENT_ID: The client ID for Google OAuth.

- - One valid Google account: Use a Google account to test the authorization process.

- - Log into multiple Google accounts: This will help test how the application handles authorization across different accounts and manage cancellations properly.
